[quant] conv2d per channel quantization support

dskhudia opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #25182 [ptquant] Minor fixes in per channel support for qconv kernel
* **#25134 [quant] conv2d per channel quantization support**

copy of #24936 

it was reverted #25131 


Per channel quantization support in qconv2d + tests

Differential Revision: [D16999104](https://our.internmc.facebook.com/intern/diff/D16999104/)

dskhudia committed:
[quant] per channel quantization support

copy of https://our.intern.facebook.com/intern/diff/D16909378/

it was reverted in https://our.intern.facebook.com/intern/diff/D16997422/


Per channel quantization support in qconv2d + tests

Differential Revision: [D16999104](https://our.internmc.facebook.com/intern/diff/D16999104/)

jianyuh commented:
This is re-applying the PR (https://github.com/pytorch/pytorch/pull/24936). Previously there is an issue of OSS landing tools: In the stack of dependent diffs, somehow #2 and #3 landed before #1. This caused #24936 lands before #24934. Previously to avoid the build error propagation, #25131 is applied.

dskhudia committed:
Update on "[quant] per channel quantization support"


copy of #24936 

it was reverted #25131 


Per channel quantization support in qconv2d + tests

Differential Revision: [D16999104](https://our.internmc.facebook.com/intern/diff/D16999104/)

dskhudia committed:
Update on "[quant] conv2d per channel quantization support"


copy of #24936 

it was reverted #25131 


Per channel quantization support in qconv2d + tests

Differential Revision: [D16999104](https://our.internmc.facebook.com/intern/diff/D16999104/)

dskhudia committed:
Update on "[quant] conv2d per channel quantization support"


copy of #24936 

it was reverted #25131 


Per channel quantization support in qconv2d + tests

Differential Revision: [D16999104](https://our.internmc.facebook.com/intern/diff/D16999104/)

dskhudia committed:
Update on "[quant] conv2d per channel quantization support"


copy of #24936 

it was reverted #25131 


Per channel quantization support in qconv2d + tests

Differential Revision: [D16999104](https://our.internmc.facebook.com/intern/diff/D16999104/)

dskhudia committed:
Update on "[quant] conv2d per channel quantization support"


copy of #24936 

it was reverted #25131 


Per channel quantization support in qconv2d + tests

Differential Revision: [D16999104](https://our.internmc.facebook.com/intern/diff/D16999104/)

dskhudia committed:
Update on "[quant] conv2d per channel quantization support"


copy of #24936 

it was reverted #25131 


Per channel quantization support in qconv2d + tests

Differential Revision: [D16999104](https://our.internmc.facebook.com/intern/diff/D16999104/)

dskhudia committed:
Update on "[quant] conv2d per channel quantization support"


copy of #24936 

it was reverted #25131 


Per channel quantization support in qconv2d + tests

Differential Revision: [D16999104](https://our.internmc.facebook.com/intern/diff/D16999104/)
